checkov/terraform/graph_builder/local_graph.py

```diff
@@ -139,6 +145,85 @@ def _add_block_data_to_graph(self, idx: int, block: TerraformBlock) -> None:
         self.in_edges[idx] = []
         self.out_edges[idx] = []
 
+    def _add_provider_attr_to_resources(self) -> None:
```

Comment:
can you add documentation to this function? looks rather complex...

Reply:
Added